Api/Resource.php

```diff
@@ -50,6 +50,34 @@ class Resource implements ResourceInterface
      */
     private $validationGroups;
     /**
+     * @var bool
+     */
+    private $paginationEnabledByDefault = false;
+    /**
+     * @var bool
+     */
+    private $clientAllowedToEnablePagination = false;
+    /**
+     * @var float
+     */
+    private $itemsPerPageByDefault = 30.;
```

Comment:
I'm not sur that displaying `2^63` items in a single request will be fine, but I see the point :)